src/Rector/AbstractRector.php

```diff
@@ -179,6 +210,41 @@ protected function isAtLeastPhpVersion(string $version): bool
         return $this->phpVersionProvider->isAtLeast($version);
     }
 
+    protected function checkCommentForIgnore(Doc $doc): bool
+    {
+        if (preg_match_all('#@noRector\s*(?<rectorName>\S+)#i', $doc->getText(), $matches)) {
+            foreach ($matches['rectorName'] as $ignoreSpec) {
+                if (static::class === ltrim($ignoreSpec, '\\')) {
+                    return true;
+                }
+            }
+        }
+
+        if ($this->isIgnoredForOtherTool($doc)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    protected function isIgnoredForOtherTool(Doc $doc): bool
+    {
+        foreach ($this->isSimilarTo() as $tool => $inspections) {
+            switch ($tool) {
+                case 'intellij':
+                    if (preg_match_all('#@noinspection\s*(?<noinspection>\S+)#i', $doc->getText(), $matches)
+                        && count(array_intersect($matches['noinspection'], $inspections))) {
+                        return true;
+                    }
+
+                    break;
+                default:
+                    throw new UnknownToolException('Unknown tool: ' . $tool);
+            }
+        }
+        return false;
+    }
```

Comment:
Looks good! Just extract the logic to external service and add via `@require` to keep `AbstractRector` prevent from being a God class